extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcTestResourceProvider.java

```diff
@@ -4,38 +4,34 @@
 import java.lang.reflect.Field;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Set;
 
-import javax.enterprise.context.spi.CreationalContext;
-import javax.enterprise.inject.spi.Bean;
-import javax.enterprise.inject.spi.BeanManager;
 import javax.inject.Inject;
 import javax.inject.Qualifier;
 
 import io.quarkus.arc.Arc;
+import io.quarkus.arc.ArcContainer;
 import io.quarkus.deployment.test.TestResourceProvider;
 
 public class ArcTestResourceProvider implements TestResourceProvider {
 
+    public static final Annotation[] EMPTY_ANNOTATION_ARRAY = new Annotation[0];
+
     @Override
     public void inject(Object test) {
         Class<?> c = test.getClass();
+        ArcContainer container = Arc.container();
         while (c != Object.class) {
             for (Field f : c.getDeclaredFields()) {
                 if (f.isAnnotationPresent(Inject.class)) {
                     try {
-                        BeanManager beanManager = Arc.container().beanManager();
```

Comment:
:+1: :+1: 

Makes this much cleaner IMHO. When looking at various pieces of the code in the past I never really understood when to use Arc and when to use the BeanManager. This makes things a lot cleaner I think!